### DISCUSSION TITLE
KeyError when trying to use installed font?

### DISCUSSION THREAD

**Discussion by Rapid1898-code** [py-pdf/pypdf#1949] [Q&A] [answered]
Hello - i installed a new font and try to use it with PyPDF2 with this line:
`pdf.setFont("Playfair Display", 10) `

But even tough the font is installed and can be seen under windows/fonts
![image](https://github.com/py-pdf/pypdf/assets/57018730/4bb7ceec-66e9-4d62-b5d4-5f0308cd92a1)

I get this error when running the program:

```
$ python workPDF.py
Working for Peter Schmidt, Musterstraße 1, 50767 Köln
Traceback (most recent call last):
  File "G:\DEV\.venv\pdf\lib\site-packages\reportlab\pdfbase\pdfmetrics.py", line 697, in getFont
    return _fonts[fontName]
KeyError: 'Playfair Display'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "G:\DEV\Fiverr\TRY\kalleinc\workPDF.py", line 54, in <module>
    pdf.setFont("Playfair Display", 10)
  File "G:\DEV\.venv\pdf\lib\site-packages\reportlab\pdfgen\canvas.py", line 1726, in setFont
    font = pdfmetrics.getFont(self._fontname)
  File "G:\DEV\.venv\pdf\lib\site-packages\reportlab\pdfbase\pdfmetrics.py", line 699, in getFont
    return findFontAndRegister(fontName)
  File "G:\DEV\.venv\pdf\lib\site-packages\reportlab\pdfbase\pdfmetrics.py", line 681, in findFontAndRegister
    face = getTypeFace(fontName)
  File "G:\DEV\.venv\pdf\lib\site-packages\reportlab\pdfbase\pdfmetrics.py", line 638, in getTypeFace
    return _typefaces[faceName]
KeyError: 'Playfair Display'

```

Why is this new font not working?

**Comment by pubpub-zz** [comment]
PyPDF2 is obsolete please upgrade to pypdf latest version
Also please provide a sample test code and possible pdf file showing your issue

**Comment by Rapid1898-code** [comment]
thanks - what is the latest version?
so instead of
`from PyPDF2 import PdfReader, PdfWriter`
i should simply use
`from pypdf import PdfReader, PdfWriter`

Correct?

**Reply by pubpub-zz** [reply]
correct but do not you have to uninstall PyPDF2 with pip and then install pypdf

**Comment by Rapid1898-code** [comment]
Hello - i have exactly the same errror-message also using pypdf

When i run the program with this statement
`pdf.setFont("Helvetica", 15)`
i have no problem - but trying to run instead with 
`pdf.setFont("Playfair Display", 10)`
i get the error message

Font is installed as you can see in the screenshot above.

**Comment by pubpub-zz** [comment, marked as answer]
pypdf does not have any setFont function. Also looking at the traces you've produce I see no traces dealing with pypdf but with reportlab.
